kyaml/fn/framework/result.go

```diff
@@ -185,3 +189,53 @@ func resultToString(item Result) string {
 	return fmt.Sprintf("resource-ref:%s,field:%s,message:%s",
 		item.ResourceRef, item.Field, item.Message)
 }
+
+func ErrorConfigFileResult(err error, path string) *Result {
+	return ConfigFileResult(err.Error(), path, Error)
+}
+
+func ConfigFileResult(msg, path string, severity Severity) *Result {
+	return &Result{
+		Message:  msg,
+		Severity: severity,
+		File: &File{
+			Path: path,
+		},
+	}
+}
+
+func ErrorResult(err error) *Result {
+	return GeneralResult(err.Error(), Error)
+}
+
+func GeneralResult(msg string, severity Severity) *Result {
+	return &Result{
+		Message:  msg,
+		Severity: severity,
+	}
+}
+
+func ErrorConfigObjectResult(err error, rn *yaml.RNode) *Result {
```

Comment:
To your point about wanting a more surface for kyaml packages, I'm not sure about these `Error___` variants being necessary. Their single-line contents look simple enough to author.